mlrun/api/initial_data.py

```diff
@@ -508,33 +541,204 @@ def _rename_marketplace_kind_to_hub(
+        new_artifact = mlrun.api.db.sqldb.models.ArtifactV2()
+
+        artifact_dict = artifact.struct
+        if is_legacy_artifact(artifact_dict):
+            # convert the legacy artifact to the new format, by setting a metadata field and spec field
+            # and copying the old fields to the spec
+            artifact_dict = mlrun.artifacts.base.convert_legacy_artifact_to_new_format(
+                artifact_dict
+            ).to_dict()
+
+        artifact_metadata = artifact_dict.get("metadata", None)
+
+        # producer_id - the current uid value
+        # uid can be in the metadata or in the artifact itself, or in the tree field
+        old_uid = artifact_metadata.get("uid", None)
+        if not old_uid:
+            old_uid = artifact_dict.get("uid", None)
+        if not old_uid:
+            old_uid = artifact_metadata.get("tree", None)
+        new_artifact.producer_id = old_uid
+
+        # uid - calculate as the hash of the artifact object
+        tag = artifact_metadata.get("tag", "")
+        uid = fill_object_hash(artifact_dict, "uid", tag)
+        new_artifact.uid = uid
+
+        # project - copy as is
+        new_artifact.project = artifact_metadata.get("project", None)
+
+        # key - the artifact's key, without iteration if it is attached to it
+        key = artifact_metadata.get("key", "")
+        new_artifact.key = key
+
+        # iteration - the artifact's iteration
+        iteration = artifact_metadata.get("iter", None)
+        if iteration is not None:
+            new_artifact.iter = int(iteration)
+
+        # kind - doesn't exist in v1, will be set to "artifact" by default
+        new_artifact.kind = artifact_dict.get("kind", mlrun.artifacts.Artifact.kind)
+
+        # updated - the artifact's updated time
+        updated = artifact_metadata.get("updated", datetime.datetime.now())
+        new_artifact.updated = updated
+
+        # created - the artifact's created time
+        # since this is a new field, we just take the updated time
+        new_artifact.created = updated
+
+        # full_object - the artifact dict
+        new_artifact.full_object = artifact_dict
+
+        # save the new object to the db
+        new_artifacts.append(new_artifact)
+
+        last_migrated_artifact_id = artifact.id
+
+        # save the artifact's tags and labels to migrate them later
+        if tag:
+            artifacts_tags_to_migrate.append((new_artifact, tag))
+        labels = artifact_metadata.get("labels", {})
+        if labels:
+            artifacts_labels_to_migrate.append((new_artifact, labels))
+
+    # commit the new artifacts to the db
+    db._upsert(db_session, new_artifacts, ignore=True)
+
+    # migrate artifact labels to the new table ("artifact_v2_labels")
+    _migrate_artifact_labels(db, db_session, artifacts_labels_to_migrate)
+
+    # migrate artifact tags to the new table ("artifact_v2_tags")
+    _migrate_artifact_tags(db, db_session, artifacts_tags_to_migrate)
```

Comment:
There's an issue here that the 1st `_upsert` will call `commit()`. Then if we fail or process crashes in one of the two migration steps happening later (tags and labels, which may still take some time), we won't be able to recover - the next time we run we will try to re-insert to the new artifacts table and will fail on conflict. 
So, I think you need to just `session.add()` here (and in the migration functions), then right before the end of this function call `commit()` once to make sure everything is committed together. 
It would then be an extremely rare thing indeed that we fail just after that commit and before we write the new migrated id to the file.